### ISSUE TITLE
Improve submission speed for many recipes

### ISSUE THREAD

**Issue by jbusecke** [pangeo-forge/pangeo-forge-runner#103]
Over at https://github.com/leap-stc/cmip6-leap-feedstock I am picking up the pace and am currently running through several hundreds of datasets a day.

I find that the main limitation for scaling this at the moment is `-runner` submitting jobs in serial. 

I am running submission from [here](https://github.com/leap-stc/cmip6-leap-feedstock/actions/runs/6112919295) and see a submitted job about every ~30 seconds:
<img width="598" alt="image" src="https://github.com/pangeo-forge/pangeo-forge-runner/assets/14314623/5a2461bf-e4aa-4a42-bd1c-1bb1b2058a49">

Is there any way to increase the submission speed here? Could we submit batches of jobs at the same time?

**Comment by yuvipanda** [comment]
We will probably have to implement a threadpool here to make concurrent submission work: https://github.com/pangeo-forge/pangeo-forge-runner/blob/35e77325d3453da9c71931ca31dbf3d05ac9b8be/pangeo_forge_runner/commands/bake.py#L223

**Comment by jbusecke** [comment]
And then maybe pass an option to limit concurrency? That would be ideal to not fail out due to quota issues.

**Comment by yuvipanda** [comment]
@jbusecke correct! I think concurrency should just be 1 by default, as that's the most common use case.
